projects/cm_mcu/CommandLineTask.c

```diff
@@ -156,62 +156,88 @@ typedef struct __attribute__((packed)) {
 
 extern struct dev_i2c_addr_t pm_addrs_dcdc[];
 
+static BaseType_t reset_all(int argc, char **argv, char *m)
+{
+  static int which = 0;
+  static int page_d = 0;
+  for (; which < N_PM_ADDRS_DCDC; ++which) {
+    for (; page_d < 2; ++page_d) { // for reading two pages per device
+      bool reset = false;
+      reset = true;
+      uint8_t sn[32];
+      snapdump(&pm_addrs_dcdc[which], page_d, sn, reset);
+    }
+  }
+  return pdFALSE;
+}
+
 static BaseType_t snapshot(int argc, char **argv, char *m)
 {
   _Static_assert(sizeof(snapshot_t) == 32, "sizeof snapshot_t");
   int copied = 0;
-  int page = strtol(argv[1], NULL, 10); // which LGA08D
-  int which = page / 10;
-  page = page % 10;
-  if (page < 0 || page > 1) {
-    copied += snprintf(m + copied, SCRATCH_SIZE - copied, "%s: page %d must be between 0-1\r\n",
-                       argv[0], page);
-    return pdFALSE;
+  if (strlen(argv[1]) < 3) {
+    char cpargv[2];
+    strcpy(cpargv, argv[1]);
+    cpargv[0] = argv[1][1];
```

Comment:
I don’t understand what these lines of code are supposed to do. Are you trying to handle the case where `strtol` doesn’t detect a number and returns zero for that? There is a correct way to do that with `errno` and the argument we set to `NULL` — did you try that? In any event, I’m not sure this is necessary.

Reply:
either way might not be necessary as you said, if we do not mix a 'reset' string command or other string commands with this `snapshot` function. Do we agree to have a new function for a reset command then? 

re: your question about what I am trying to do here 
yes, I was trying to avoid the return zero of any non-numerial inputs. I found [an example online](https://wiki.sei.cmu.edu/confluence/display/c/ERR34-C.+Detect+errors+when+converting+a+string+to+a+number) for the case when we use `errno.h.` So if we need this kind of detection, I will do it this way instead. Thank  you.

Reply:
Done.